Routes/Question/Qsts.js

```diff
@@ -58,135 +69,101 @@ router.post('/', function(req, res) {
+// delete specific question if AU is owner or admin
+router.delete('/:qstId', function(req, res) {
    var vld = req.validator;
-   var cnvId = req.params.cnvId;
+   var qstId = req.params.qstId;
    var cnn = req.cnn;
-   var query = 'select Message.id, UNIX_TIMESTAMP(whenMade) * 1000' +
-    ' as whenMade, email, content from Conversation c join' +
-    ' Message on cnvId = c.id join Person p on prsId = p.id where (c.id = ?' +
-    ' and UNIX_TIMESTAMP(whenMade) * 1000 < ?) order by whenMade asc, id';
-   var d = new Date();
-   var date = req.query.hasOwnProperty("dateTime") ? 
-    parseInt(req.query.dateTime) : d.getTime();
-   var params = [parseInt(cnvId), date];
-
-   // And finally add a limit clause and parameter if indicated.
-   if (req.query.hasOwnProperty("num")) {
-      query += ' limit ?';
-      params.push(parseInt(req.query.num));
-   }
 
    async.waterfall([
-   function(cb) {  // Check for existence of conversation
-      cnn.chkQry('select * from Conversation where id = ?', [cnvId], cb);
-   },
-   function(cnvs, fields, cb) { // Get indicated messages
-      if (vld.check(cnvs.length, Tags.notFound, null, cb))
-         cnn.chkQry(query, params, cb);
+   function(cb) {
+      cnn.chkQry('select * from Question where id = ?', qstId, cb);
    },
-   function(msgs, fields, cb) { // Return retrieved messages
-      res.json(msgs);
-      cnn.release();
+   function(qsts, fields, cb) {
+      if(vld.check(qsts.length, Tags.notFound, null, cb) &&
+       vld.checkPrsOK(qsts[0].id, cb)) {
+         cnn.chkQry('delete from Question where id = ?', qstId, cb);
+      }
    }],
    function(err){
+      if(!err) {
+         res.status(200).end();
+      }
       cnn.release();
    });
 });
 
-router.post('/:cnvId/Msgs', function(req, res){
+// submit guess to question, if correct add user to PersonQuestion table
+router.post('/:qstId/Answers', function(req, res) {
    var vld = req.validator;
    var cnn = req.cnn;
-   var cnvId = req.params.cnvId;
-   var now;
+   var body = req.body;
+   var qstId = req.params.qstId;
 
    async.waterfall([
    function(cb) {
-      if(vld.check(req.body.hasOwnProperty("content"), Tags.missingField,
-       ["content"], cb) && vld.check(req.body.content.length < 5000, 
-       Tags.badValue, ["content"], cb)) {
-         cnn.chkQry('select * from Conversation where id = ?', [cnvId], cb);
-      }
+      //get question
+      cnn.chkQry('select * from Question where id = ?', qstId, cb);
    },
-   function(cnvs, fields, cb) {
-      if (vld.check(cnvs.length, Tags.notFound, null, cb)) {
-         cnn.chkQry('insert into Message set ?',
-          {cnvId: cnvId, prsId: req.session.id,
-          whenMade: now = new Date(), content: req.body.content}, cb);
+   function(qsts, fields, cb) {
+      //check if question exists
+      if(vld.check(qsts.length, Tags.notFound, cb) &&
+       vld.hasFields(body, ["answer"], cb) &&
+       vld.check(qsts[0].answer === body.answer, null, null, cb)) {
```

Comment:
should probably not use vld.check here, otherwise user will get a 400 error if they put in the wrong answer (And I think they should get a 200 right?)